x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichCache.java

```diff
@@ -89,6 +90,7 @@ private Cache<CacheKey, CacheValue> createCache(long maxWeight, ToLongBiFunction
      * @param searchResponseFetcher The function used to compute the value to be put in the cache, if there is no value in the cache already
      * @param listener A listener to be notified of the value in the cache
      */
+    @FixForMultiProject(description = "The enrich cache will currently leak data between projects. We need to either disable or fix it.")
```

Comment:
Seems pretty serious if it happens, I suggest we create a JIRA issue to track it for better visibility.

Reply:
I already created [ES-10936](https://elasticco.atlassian.net/browse/ES-10936) and put it on the agenda for today's weekly Data Management team meeting :)